### PR TITLE
Policy Mangement Service

### PR DESCRIPTION
**Pull request by setrofim**
Initial implementation of policy management REST API, implemented as a separate service node.

### CODE REVIEW

File: integration-tests/Makefile [hunk 1]
```diff
@@ -23,7 +23,8 @@ $(DEPLOYMENT_SRC_DIR).built/%:
 
 
 CONTAINER_FLAGS := --env-file  $(DEPLOYMENT_SRC_DIR)deployment.cfg --network veraison-net \
-		   -v $(THIS_DIR):/integration-tests
+		   -v $(THIS_DIR):/integration-tests \
+		   -v $(STORES_VOLUME):/opt/veraison/stores
```
Comment (yogeshbdeshpande) [inline integration-tests/Makefile:27]: Question on line 29:

a) Where is `DEPLOYMENT_DEPS` used ?

b) If used, should we not append the management-container

Comment (setrofim) [reply integration-tests/Makefile:27]: They're not, good catch. I think the intiention was for them to be used as the dependencies for the `.built/image` rule, but uses `$(DEPLOYMENT_SRC_DIR).built/deploy` instead, which encapsulates those (and would cover the new binary as well).

I'll remove.

**Comment by yogeshbdeshpande** [comment]
> Added a commit replacing `polcli` with `pocli`.

Thanks!

File: policy/cmd/polcli/commands/add.go [hunk 2]
```diff
@@ -46,7 +46,11 @@ func validateAddArgs(cmd *cobra.Command, args []string) error {
 }
 
 func doAddCommand(cmd *cobra.Command, args []string) error {
-	policyID := args[0]
+	policyID, err := policy.PolicyKeyFromString(args[0])
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply policy/cmd/polcli/commands/add.go]: This has already been addressed in the current PR so issue is resolved.

File: management/cmd/management-service/config.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  backend: go-plugin
+  go-plugin:
+    dir: ../../../scheme/bin/
+po-store:
+  backend: sql
+  sql:
+    driver: sqlite3
+    datasource: /veraison/stores/vts/po-store.sql
+management:
+  listen-addr: 0.0.0.0:8088
```
[the remark this answers is not in the document]
Comment (setrofim) [reply management/cmd/management-service/config.yaml:11]: This is the address the service is listening on. I.e. which connections it accepts the  `0.0.0.0:8088` means "listen on port 8088 for connection from any host".

File: management/policy.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+	}
+
+	store, err := policy.NewStore(subs["po-store"], log.Named(name+"-store"))
+	if err != nil {
+		return nil, err
+	}
+
+	var pluginManager plugin.IManager[handler.IEvidenceHandler]
+
+	if config.SchemeLoader == "plugins" { // nolint:gocritic
```
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply management/policy.go:44]: issue #178 tracks this item

File: policy/policy_key.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// PolicyKey identifies a specific policy. This is used to retrieve the policy
+// from the store.
+type PolicyKey struct {
```
[the remark this answers is not in the document]
Comment (setrofim) [reply policy/policy_key.go:13]: Added: https://github.com/veraison/services/issues/179

File: management/policy.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (setrofim) [reply management/policy.go:44]: Added: https://github.com/veraison/services/issues/178

File: policy/policy_key.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (yogeshbdeshpande) [reply policy/policy_key.go:13]: Awesome!

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM!
Thanks for the changes.

**Pull request merged**